### DISCUSSION TITLE
Unable to classify C++ repo correctly

### DISCUSSION THREAD

**Discussion by antoan** [github-linguist/linguist#5793] [Classification] [answered]
Repo in question: https://github.com/antoan/nemesis
Here is what I have tried so far: 

I have added a .gitattributes to the repo with the following content as a quick test:

```
*.h linguist-language=C++
*.cpp linguist-language=C++

```

https://github.com/antoan/nemesis/blob/master/.gitattributes

The C++ source can be found the following sub directory:

https://github.com/antoan/nemesis/tree/master/nemesis_base 

I cannot tell if I have done things incorrectly or just that linguist requires some time to process the repo after I have commited?

Any help would be much appreciated.

Cheers

**Comment by lildude** [comment, marked as answer]
🤔 I'm not sure why you think you need the overrides. From a quick look at your repo, Linguist will probably classify correctly without any nudging.

So I tested this, and yup, no probs:

```console
➜ gh repo clone antoan/nemesis
Cloning into 'nemesis'...
remote: Enumerating objects: 631, done.
remote: Counting objects: 100% (631/631), done.
remote: Compressing objects: 100% (268/268), done.
remote: Total 631 (delta 360), reused 580 (delta 327), pack-reused 0
Receiving objects: 100% (631/631), 1.61 MiB | 2.88 MiB/s, done.
Resolving deltas: 100% (360/360), done.
➜ cd nemesis/
➜ github-linguist .
45.89%  35836      Python
33.88%  26457      C++
20.24%  15804      CMake
➜ git rm .gitattributes
rm '.gitattributes'
➜ git commit -m 'Remove gitattributes'
[master 60f2b42] Remove gitattributes
 1 file changed, 13 deletions(-)
 delete mode 100644 .gitattributes
➜ github-linguist .
45.89%  35836      Python
33.88%  26457      C++
20.24%  15804      CMake

```

As you can see, no difference with or without your override.

This also matches what is currently shown in the sidebar of your repo.

**Reply by antoan** [reply]
My bad, I should have clarified that when viewing my pinned repositories cards - only Python is listed?

**Reply by Nixinova** [reply]
Because python is the language with the most content in the repo.

**Reply by antoan** [reply]
I see, much appreciated, thank you.
